### ISSUE TITLE
Tamil Language

### ISSUE THREAD

**Issue by arulrajnet** [JaidedAI/EasyOCR#39]
Added support for Tamil language 

https://en.wikipedia.org/wiki/Tamil_script

**Comment by arulrajnet** [comment]
Here is the PR #37

**Comment by rkcosmos** [comment]
Ok, waiting for word list.

**Comment by Vijayabhaskar96** [comment]
I would like to know how to add a new language to the repo and get it working, For example: the character and words for the Tamil language is now added to the repo but how to get it working? how to train the model?

**Comment by rkcosmos** [comment]
I'll train the model and update the repo. It usually takes me at least one week for each model.

**Comment by rkcosmos** [comment]
I'm now trying to do Tamil OCR. The problem is rendering something like 'கௌ'. Now I'm having
![image](https://user-images.githubusercontent.com/12238328/87385855-5ac24180-c5c9-11ea-83cf-e77cab8ef849.png)
image from Pillow. How can I render it correctly?

**Comment by rkcosmos** [comment]
from https://en.wikipedia.org/wiki/Tamil_All_Character_Encoding. Looks like I cannot use Unicode with Tamil

**Comment by Vijayabhaskar96** [comment]
I used [open-tamil](https://pypi.org/project/Open-Tamil/) to convert unicode to bamini font and rendered using the same font and it worked.
```
from PIL import Image
from PIL import ImageFont, ImageDraw
image=Image.new("RGB",[320,320])
draw = ImageDraw.Draw(image)
a="கௌ"
a=tamil.txt2unicode.unicode2bamini(a)
font=ImageFont.truetype("F:/Bamini.ttf",25)
draw.text((50, 50), a, font=font)
image```

you can download the font here https://www.freetamilfont.com/download.php?id=735930

**Comment by rkcosmos** [comment]
it works! Is there other font that work like this? I normally use >20 fonts for other languages? It's important to have variation. I already have a lot of unicode font but not a single one works.

**Comment by vanangamudi** [comment]
> from https://en.wikipedia.org/wiki/Tamil_All_Character_Encoding. Looks like I cannot use Unicode with Tamil

Though TACE16 is well designed encoding for Tamil, UTF-8 is the most popular encoding for Tamil that is in wide spread use. There are few unicode fonts for Tamil. We have to use complex layout engines like pango for eastern languages like Tamil. 

Let me try this out

**Comment by rkcosmos** [comment]
Do you guys want Tamil's number in the model as well. https://en.wikipedia.org/wiki/Tamil_numerals
They looks similar to what is in the list but python says they are different. or is it just unicode issue?

**Comment by Vijayabhaskar96** [comment]
No, I don't. Nobody uses them.

**Comment by rkcosmos** [comment]
ok, please tell me if there's any clue about rendering, I am in blind spot now. Do u guys use unicode font in other software, like word, excel, notepad, etc? I found old post complaining about rendering in windows7, 10.

**Comment by Vijayabhaskar96** [comment]
I'm trying to find answers, so far I think PIL is what messing up the order of the characters while rendering. Is there any other way to render texts as image?

**Comment by Vijayabhaskar96** [comment]
Did you figure out a way to render? @vanangamudi told me to try 
> `convert pango:"தமிழ்" output.gif`
We need to install imagemagick and pango. I am on ubuntu. any linux machine should work.

I don't have a Linux machine, tried on colab but had installation issues. can you give a try if nothing works?

**Comment by rkcosmos** [comment]
I'm on Linux, will try. I think we have no choice but using other programming languages to render. It is surely possible but I have to modify my training pipeline a lot, so I'll put Tamil on hold for a while and work on other languages first. Will come back to Tamil later, everything should be done within Roadmap's timeline.

**Comment by Vijayabhaskar96** [comment]
@rkcosmos Good news! Finally, I found a way to render all Tamil characters correctly with Unicode fonts.
pyvips renders everything fine(I rendered and glanced every alphabet in ta_cha.txt manually and everything looks fine).
Remember to use unique names for the font while loading, I got the same font even after loading next font(maybe because I was testing on Jupiter notebook the states were saved?)
```
import pyvips
image = pyvips.Image.text(text_string, dpi=3000, font='Catamaran', \
                          fontfile="F:/tamil-fonts-master/open-sourced/Catamaran-Black.ttf")
image.write_to_file("text.png")
```
